geopandas/geodataframe.py

```diff
@@ -631,6 +631,31 @@ def _dataframe_set_geometry(self, col, drop=False, inplace=False, crs=None):
                          " DataFrame to GeoDataFrame")
     return GeoDataFrame(self, geometry=col, crs=crs)
 
+
+def concat(L, axis=0):
+    """ Concatenate multiple GeoDataFrames or GeoSeries together
+
+    Currently only works for axis=0
+    """
+    if axis != 0:
+        raise NotImplementedError("Only axis=0 supported")
+
+    if isinstance(L[0], GeoDataFrame):
+        name = L[0]._geometry_column_name
+    else:
+        name = L[0].name
+
+    geometry = vectorized.concat(df._geometry_array for df in L)
+    if isinstance(L[0], GeoDataFrame):
+        L = [df.drop(name, axis=1) for df in L]
+        new = pd.concat(L)
+        new = new.set_geometry(GeoSeries(geometry, name=name, index=new.index))
+    else:
+        index = pd.concat([df.to_frame().drop(name, axis=1) for df in L]).index
```

Comment:
you can also do `pd.concat([s.index.to_series() for s in L]).index`

Reply:
Is this preferred?  It made more sense to me to use `to_frame` in order to get the index.

Reply:
Not necessarly, I thought it was a bit shorter.

But from a quick test, it seems faster than the to_frame method

Reply:
You could also do [pd.Series(index=s.index) for s in L]